### PR TITLE
docs(blob): update godoc comments to better match what Get and GetProof actually do

### PR DESCRIPTION
**Pull request by ramin**
Refs #3335 

Attempts to create more accurate doc/comments on Get and GetProof in `blob/service.go` to improve legibility and comprehension of how these methods behave to API consumers

specifically cc @vgonkivs

### CODE REVIEW

File: blob/service.go [hunk 1]
```diff
@@ -136,9 +137,9 @@ func (s *Service) Get(
 	return
 }
 
-// GetProof retrieves all blobs in the given namespaces at the given height by commitment
-// and returns their Proof. It collects all namespaced data from the EDS, constructs blobs
-// and compares commitments.
+// GetProof retrieves a blob in the given namespace at the given height by commitment
+// and returns their Proof. It collects all namespaced data from the EDS, constructs the blob
+// and compares the commitment argument.
```
[the remark this answers is not in the document]
Comment (ramin) [reply blob/service.go:142]: @walldiss wanna make an inline suggestion of how it should look and we can accept based on that?

Comment (walldiss) [inline blob/service.go:142]:
```suggestion
// GetProof returns an NMT inclusion proof for a blob within the specified namespace at the given
// height, using the given commitment. It employs the same algorithm as service.Get() internally.
```

**Review by renaynay — APPROVED** [review]

File: blob/service.go [hunk 2]
```diff
@@ -137,9 +138,8 @@ func (s *Service) Get(
 	return
 }
 
-// GetProof retrieves all blobs in the given namespaces at the given height by commitment
-// and returns their Proof. It collects all namespaced data from the EDS, constructs blobs
-// and compares commitments.
+// GetProof returns an NMT inclusion proof for a blob within the specified namespace at the given
```
Comment (vgonkivs) [inline blob/service.go:141]: Let's also specify that the Proof itself is not a Blob inclusion Proof but the namespace inclusion proof to the respective row roots.

Comment (ramin) [reply blob/service.go:141]: @vgonkivs can you draft a suggested edit?

Comment (vgonkivs) [reply blob/service.go:141]: `// GetProof returns an NMT inclusion proof for a specified namespace to the respective row roots on which the blob spans on`

**Review by vgonkivs — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
thanks for clarification on the proof

**Pull request merged**